### PR TITLE
(main) Release: 2.6.0

### PR DESCRIPTION
**Pull request by Adamj1232**
### ✨ Highlights

-  Fix: Autoselect with disabled modals resolve #1154 
-  Fix: Chain id type fix #1158 
-  Enhancement: Minify bundle size #1155 
-  Fix: add "main" to Keystone's package.json #1164 
-  Fix Trezor nonce handling #1165 
-  Fix - Add optionality to containerElement, format validation so accountCener #1166 
-  Feature : Gas Module #1168 
-  Fix - Updated Binance Wallet provider patch #1174 
-  Enhancement : Convert internal Wei to Eth #1137 
-  Feature : Add BitKeep injected Wallet #1177 

### 📦 Changes per package

- `@web3onboard/coinbase` **v2.0.10**
- `@web3onboard/common` **v2.1.7**
- `@web3onboard/core` **v2.6.0**
- `@web3onboard/dcent` **v2.0.7**
- `@web3onboard/formatic` **v2.0.9**
- `@web3onboard/gas` **v2.0.0**
- `@web3onboard/gnosis` **v2.0.8**
- `@web3onboard/injected` **v2.0.15**
- `@web3onboard/keepkey` **v2.1.7**
- `@web3onboard/keystone` **v2.1.8**
- `@web3onboard/ledger` **v2.1.7**
- `@web3onboard/magic` **v2.0.10**
- `@web3onboard/mew` **v2.0.7**
- `@web3onboard/portis` **v2.0.7**
- `@web3onboard/react` **v2.2.5**
- `@web3onboard/torus` **v2.0.8**
- `@web3onboard/trezor` **v2.1.7**
- `@web3onboard/vue` **v2.1.5**
- `@web3-onboard/walletconnect` **v2.0.8**
- `@web3onboard/web3auth` **v2.0.6**

### Checklist
- [ ] The version field in `package.json` is incremented following [semantic versioning](https://semver.org/)
- [ ] The box that allows repo maintainers to update this PR is checked
- [ ] I tested locally to make sure this feature/fix works
- [ ] I have run `yarn type-check` & `yarn build` to confirm there are not any associated errors
- [ ] This PR passes the Circle CI checks

### CODE REVIEW

File: packages/injected/src/wallets.ts [hunk 1]
```diff
@@ -78,31 +83,22 @@ const binance: InjectedWalletModule = {
...
-          return accts
-        }),
-      eth_selectAccounts: UNSUPPORTED_METHOD,
       eth_chainId: ({ baseRequest }) =>
-        baseRequest({ method: 'eth_chainId' }).then(
-          id => `0x${parseInt(id).toString(16)}`
-        ),
+        baseRequest({ method: 'eth_chainId' }).then(id => {
+          return `0x${parseInt(id as string).toString(16)}`
+        }),
```
Comment (taylorjdawson) [inline packages/injected/src/wallets.ts:99]:
```suggestion
        baseRequest({ method: 'eth_chainId' }).then(id => `0x${parseInt(id as string).toString(16)}`),
```

**Review by aaronbarnardsound — APPROVED** [review]

File: packages/core/src/validation.ts [hunk 2]
```diff
@@ -305,7 +305,7 @@ export function validateString(str: string, label?: string): ValidateReturn {
 }
 
 export function validateSetChainOptions(data: {
-  chainId: ChainId
+  chainId: ChainId | DecimalChainId
```
Comment (taylorjdawson) [inline packages/core/src/validation.ts:308]: if change the types to `type ChainId = number | string` then we could have just `ChainId` here

File: packages/core/src/utils.ts [hunk 3]
```diff
@@ -92,6 +93,16 @@ export async function copyWalletAddress(text: string): Promise<void> {
   }
 }
 
+export const toHexString = (val: number | string): string => typeof val === 'number' ? `0x${val.toString(16)}` : val
+
+export function chainIdToHex(chains : Chain[] | ChainWithDecimalId[] ):
```
Comment (taylorjdawson) [inline packages/core/src/utils.ts:98]: if change the types to `type ChainId = number | string` then we could have just `Chain[]` vs `Chain[] | ChainWithDecimalId[]`

File: packages/core/src/types.ts [hunk 4]
```diff
@@ -21,7 +24,7 @@ export interface InitOptions {
   /**
    * The chains that your app works with
    */
-  chains: Chain[]
+  chains: Chain[] | ChainWithDecimalId[]
```
Comment (taylorjdawson) [inline packages/core/src/types.ts:27]: if change the types to `type ChainId = number | string` then we could have just `Chain[]` vs `Chain[] | ChainWithDecimalId[]`

**Review by taylorjdawson — APPROVED** [review]
approved with nits -- can be applied to the release or in a separate PR

File: packages/injected/src/wallets.ts (hunk 1 above)
Comment (Adamj1232) [reply packages/injected/src/wallets.ts]: Good catch! Thanks! Fixed on branch for formatting

File: packages/common/rollup.config.js [hunk 5]
```diff
@@ -33,6 +35,18 @@ export default {
     typescript({
       sourceMap: !production,
       inlineSources: !production
+    }),
+    production && terser({
```
[the remark this answers is not in the document]
Comment (mahmud-bn) [reply packages/common/rollup.config.js:39]: @taylorjdawson Yes, we are. For core too.

File: packages/trezor/src/index.ts [hunk 6]
```diff
@@ -477,7 +486,7 @@ function trezor(options: TrezorOptions): WalletInit {
               ? [accounts[0].address]
               : [],
           eth_chainId: async () =>
-            currentChain.hasOwnProperty('id') ? currentChain.id : '',
+            currentChain.hasOwnProperty('id') ? String(currentChain.id) : '',
```
[the remark this answers is not in the document]
Comment (Adamj1232) [reply packages/trezor/src/index.ts]: Fixed!

**Review by aaronbarnardsound — APPROVED** [review]

**Review by mahmud-bn — APPROVED** [review]

**Pull request merged**